### PR TITLE
Bnpl payment integration

### PR DESCRIPTION
**Pull request by deryacakmak**

### CODE REVIEW

**Review by AlicanAkkus — APPROVED** [review]

File: samples/payment/BnplPaymentInit.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            quantity: 2,
+        },
+        {
+            id: "100",
+            name: "Test Elektronik 1",
+            brandName:"Samsung",
+            type: Craftgate.Model.CardItemType.MobilePhoneOver5000Try,
+            unitPrice: 7000,
+            quantity: 1,
+        }
```
Comment (MasterSlave) [inline samples/payment/BnplPaymentInit.js:39]: items ve cartItems ı request in en altına taşıyabiliriz.

File: src/adapter/PaymentAdapter.ts [hunk 2]
```diff
@@ -181,6 +185,18 @@ export default class PaymentAdapter extends BaseAdapter {
     return this._client.post(`/payment/v1/apple-pay/merchant-sessions`, request);
   }
 
+  async offers(request: BnplPaymentOfferRequest): Promise<BnplPaymentOfferResponse> {
```
Comment (MasterSlave) [inline src/adapter/PaymentAdapter.ts:188]: retrieveBnplOffers

File: src/adapter/PaymentAdapter.ts [hunk 3]
```diff
@@ -181,6 +185,18 @@ export default class PaymentAdapter extends BaseAdapter {
...
+    return this._client.post(`/payment/v1/bnpl-payments/offers`, request);
+  }
+
+  async init(request: BnplPaymentInitRequest): Promise<BnplPaymentInitResponse> {
```
Comment (MasterSlave) [inline src/adapter/PaymentAdapter.ts:192]: initBnplPayment

File: src/adapter/PaymentAdapter.ts [hunk 4]
```diff
@@ -181,6 +185,18 @@ export default class PaymentAdapter extends BaseAdapter {
...
+    return this._client.post(`/payment/v1/bnpl-payments/init`, request);
+  }
+
+  async approve(paymentId: number): Promise<void> {
```
Comment (MasterSlave) [inline src/adapter/PaymentAdapter.ts:196]: approveBnplPayment

File: src/model/CartItemType.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  Shoes = 'SHOES',
+  Clothing = 'CLOTHING',
+  CosmeticsAndPersonalCare = 'COSMETICS_AND_PERSONAL_CARE',
+  Furniture = 'FURNITURE',
+  HomeLiving = 'HOME_LIVING',
+  AutomobileMotorcycle = 'AUTOMOBILE_MOTORCYCLE',
+  Other = 'OTHER'
+}
+
+export default CardItemType;
```
Comment (MasterSlave) [inline src/model/CartItemType.ts:28]: BnplCartItemType

File: src/model/BnplCardItemType.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+enum BnplCardItemType {
```
Comment (MasterSlave) [inline src/model/BnplCardItemType.ts:1]:
```suggestion
enum BnplCartItemType {
```

File: src/adapter/PaymentAdapter.ts [hunk 7]
```diff
@@ -181,6 +185,18 @@ export default class PaymentAdapter extends BaseAdapter {
     return this._client.post(`/payment/v1/apple-pay/merchant-sessions`, request);
   }
 
+  async retrieveBnplOffers(request: OfferBnplPaymentRequest): Promise<BnplPaymentOfferResponse> {
```
Comment (MasterSlave) [inline src/adapter/PaymentAdapter.ts:188]:
```suggestion
  async retrieveBnplOffers(request: BnplPaymentOfferRequest): Promise<BnplPaymentOfferResponse> {
```

**Review by MasterSlave — APPROVED** [review]

**Pull request merged**